cassandra-executor/src/main/java/com/mesosphere/dcos/cassandra/executor/tasks/UpgradeSSTable.java

```diff
@@ -0,0 +1,118 @@
+ */
+public class UpgradeSSTable implements ExecutorTask {
+    private static final Logger LOGGER = LoggerFactory.getLogger(UpgradeSSTable.class);
+
+    private final CassandraDaemonProcess daemon;
+    private final ExecutorDriver driver;
+    private final UpgradeSSTableTask task;
+
+    private List<String> getKeySpaces() {
+        if (task.getUpgradeSSTableContext().getKeySpaces().isEmpty()) {
+            return daemon.getNonSystemKeySpaces();
+        } else {
+            return task.getUpgradeSSTableContext().getKeySpaces();
+        }
+    }
+
+    private List<String> getColumnFamilies() {
+        return task.getUpgradeSSTableContext().getColumnFamilies();
+    }
+
+    private void sendStatus(ExecutorDriver driver,
+                            Protos.TaskState state, String message) {
+        Protos.TaskStatus status =
+            task.createStatus(state, Optional.of(message)).getTaskStatus();
+        driver.sendStatusUpdate(status);
+    }
+
+    /**
+     * Construct a new UpgradeSSTable.
+     *
+     * @param driver The ExecutorDriver used to send task status.
+     * @param daemon The CassandraDaemonProcess used to cleanup the node.
+     * @param task   The UpgradeSSTableTask executed by the UpgradeSSTable.
+     */
+    public UpgradeSSTable(final ExecutorDriver driver,
+                          final CassandraDaemonProcess daemon,
+                          final UpgradeSSTableTask task) {
+        this.driver = driver;
+        this.daemon = daemon;
+        this.task = task;
+    }
+
+    @Override
+    public void run() {
+        try {
+
+            final List<String> keySpaces = getKeySpaces();
+            final List<String> columnFamilies = getColumnFamilies();
+            sendStatus(driver, Protos.TaskState.TASK_RUNNING,
+                String.format("Starting upgradesstable: keySpaces = %s, " +
+                        "columnFamilies = %s",
+                    keySpaces,
+                    columnFamilies));
+
+            for (String keyspace : keySpaces) {
+                LOGGER.info("Starting upgradesstable : keySpace = {}, " +
+                        "columnFamilies = {}",
+                    keyspace,
+                    Arrays.asList(columnFamilies));
+
+                daemon.upgradeSSTables(keyspace, columnFamilies);
+
+                LOGGER.info("Completed upgradesstable : keySpace = {}, " +
+                        "columnFamilies = {}",
+                    keyspace,
+                    Arrays.asList(columnFamilies));
+            }
+
+            sendStatus(driver, Protos.TaskState.TASK_FINISHED,
+                String.format("Completed upgradesstable: keySpaces = %s, " +
+                        "columnFamilies = %s",
+                    keySpaces,
+                    Arrays.asList(columnFamilies)));
+        } catch (final Throwable t) {
+            LOGGER.error("UpgradeSSTable failed", t);
+            sendStatus(driver, Protos.TaskState.TASK_FAILED, t.getMessage());
```

Comment:
do we want to clean ZK status in case of failure or keep it there for next retry?